index.test.js

```diff
@@ -8,53 +8,183 @@ describe('[Exercise 1] trimProperties', () => {
+    const copy = utils.trimPropertiesMutation(input)
+    expect(copy).toEqual(input)
+    expect(copy).toBe(input)
+  }
+  )
 })
 
 describe('[Exercise 3] findLargestInteger', () => {
-  // test('[5] returns the largest number in an array of objects { integer: 2 }', () => {})
+  test('[5] returns the largest number in an array of objects { integer: 2 }', () => {
+    const array = [
+      {integer: 10}, 
+      {integer: 543}, 
+      {integer: 93}, 
+      {integer: 0}, 
+      {integer: -20}, 
+      {integer: 6}
+    ]
+    const result = utils.findLargestInteger(array)
+    expect(result).toBe(543)
+  })
 })
 
 describe('[Exercise 4] Counter', () => {
   let counter
   beforeEach(() => {
     counter = new utils.Counter(3) // each test must start with a fresh couter
   })
-  // test('[6] the FIRST CALL of counter.countDown returns the initial count', () => {})
-  // test('[7] the SECOND CALL of counter.countDown returns the initial count minus one', () => {})
-  // test('[8] the count eventually reaches zero but does not go below zero', () => {})
+
+  test('[6] the FIRST CALL of counter.countDown returns the initial count', () => {
+    const firstCount = counter.countDown()
+    expect(firstCount).toBe(3)
+  })
+
+  test('[7] the SECOND CALL of counter.countDown returns the initial count minus one', () => {
+    counter.countDown()
+    const secondCount = counter.countDown()
+    expect(secondCount).toBe(2)
+  })
+
+  test('[8] the count eventually reaches zero but does not go below zero', () => {
+    counter.countDown()
+    counter.countDown()
+    counter.countDown()
+    const fourthCount = counter.countDown()
+    const fifthCount = counter.countDown()
+
+    expect(fourthCount).toBe(0)
+    expect(fourthCount).not.toBeLessThan(0)
+    expect(fifthCount).not.toBeLessThan(0)
+    expect(fifthCount).toBe(0)
+  })
 })
 
 describe('[Exercise 5] Seasons', () => {
   let seasons
   beforeEach(() => {
     seasons = new utils.Seasons() // each test must start with fresh seasons
   })
-  // test('[9] the FIRST call of seasons.next returns "summer"', () => {})
-  // test('[10] the SECOND call of seasons.next returns "fall"', () => {})
-  // test('[11] the THIRD call of seasons.next returns "winter"', () => {})
-  // test('[12] the FOURTH call of seasons.next returns "spring"', () => {})
-  // test('[13] the FIFTH call of seasons.next returns again "summer"', () => {})
-  // test('[14] the 40th call of seasons.next returns "spring"', () => {})
+
+  test('[9] the FIRST Count of seasons.next returns "summer"', () => {
+    const firstSeason = seasons.next()
+    expect(firstSeason).toBe('summer')
+  })
+
+  test('[10] the SECOND call of seasons.next returns "fall"', () => {
+    const firstSeason = seasons.next()
+    const secondSeason = seasons.next()
+    expect(secondSeason).toBe('fall')
+  })
+
+  test('[11] the THIRD call of seasons.next returns "winter"', () => {
+    const firstSeason = seasons.next()
```

Comment:
Same as above. If you aren't using the variables later you don't need to set them